### PR TITLE
Q-001 Disconnection flow

### PR DESCRIPTION
**Pull request by daart**
- added player disconnection for real after 10sec
- added list of disconnected players on shitty UI
- and denchik lokh! Yes?

### CODE REVIEW

File: server/src/index.ts [hunk 1]
```diff
@@ -48,23 +54,48 @@ io.on("connection", function(socket) {
...
 
   socket.on("disconnect", () => {
+    // onPlayerDisconnect
     const p = playerBySocket(socket.id);
 
     if (!p) {
       return;
     }
+    
+    const game = gameByPlayer(p.id);
```
Comment (bigjey) [inline server/src/index.ts]: what if player is not in game?

File: server/src/index.ts [hunk 2]
```diff
@@ -48,23 +54,48 @@ io.on("connection", function(socket) {
...
+    
+    const timerID = setTimeout(() => {
+      console.log(`player ${p.id} was disconnected from game`);
+
+      clearTimeout(_disconnections[p.id]);
```
Comment (bigjey) [inline server/src/index.ts]: why would you clear timeout from inside of it's callback?

File: server/src/index.ts [hunk 3]
```diff
@@ -48,23 +54,48 @@ io.on("connection", function(socket) {
...
+      delete _disconnections[p.id];
+
+      game.removePlayerFromGame(p.id);
+    }, 10000);
```
Comment (bigjey) [inline server/src/index.ts]: no magic numbers?

File: server/src/index.ts (hunk 1 above)
Comment (daart) [reply server/src/index.ts]: than player will suck someone's dick hahaha

File: server/src/index.ts (hunk 2 above)
Comment (daart) [reply server/src/index.ts]: best practices in da house

File: server/src/index.ts (hunk 3 above)
Comment (daart) [reply server/src/index.ts]: nit magic numbers, especially on PORT

**Pull request merged**